extensions/amp-app-banner/0.1/amp-app-banner.js

```diff
@@ -472,18 +474,16 @@ export class AmpAndroidAppBanner extends AbstractAppBanner {
       return;
     }
 
-    for (let i = 0; i < apps.length; i++) {
-      const app = apps[i];
-      if (app['platform'] == 'play') {
-        const installAppUrl = `https://play.google.com/store/apps/details?id=${app['id']}`;
-        const openInAppUrl = this.getAndroidIntentForUrl_(app['id']);
-        this.setupOpenButton_(
-          dev().assertElement(this.openButton_),
-          openInAppUrl,
-          installAppUrl
-        );
-        return;
-      }
+    const playApp = apps.find((a) => a['platform'] === 'play');
+    if (playApp) {
+      const installAppUrl = `https://play.google.com/store/apps/details?id=${playApp['id']}`;
+      const openInAppUrl = this.getAndroidIntentForUrl_(playApp['id']);
+      this.setupOpenButton_(
+        dev().assertElement(this.openButton_),
+        openInAppUrl,
+        installAppUrl
+      );
+      return;
```

Comment:
Just out of curiosity, are there specific benefits to refactoring the 0.1 AMP component, or is it just better syntax?

Reply:
After copying this code to `1.0`, we decided to improve the syntax, and so I backported that change to the old component, to maintain the symmetry.
It sure would be nice to extract most of this logic into a common place, but it doesn't seem like any of the other Bento components share any code, so I guess having 2 copies that are very similar is the best for now.